Consume ui5lab library as project dependency

BenReim opened the pull request:
- Remove postInstall script
- Add files for ui5 self-contained build
- Pull up dependencies

BenReim committed:
Consume ui5lab library as project dependency
Add files for ui5 self-contained build

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/UI5Lab/UI5Lab-app-simple?pullRequest=7) <br/>All committers have signed the CLA.

Michadelic commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+node_modules
+dist
+webapp/thirdparty
```
i see you removed the postInstall step - this line can thus be removed

Michadelic committed:
Switched to declarative component loading to get rid of inline code and the Shell dependency in index.html
Added build scripts to package.json
Renamed self-contained index file to camel case

TODO: library CSS is not loaded properly

Michadelic committed:
removed unused folder from gitignore

Michadelic committed:
typo and indentation fix

Michadelic commented on webapp/index-self-contained.html:
```diff
@@ -0,0 +1,32 @@
+		<title>GeometryApp</title>
+
+		<script id="sap-ui-bootstrap"
+			src="resources/sap-ui-custom.js"
```
would be nice to provide two build scripts in in the package.json.
One for self-contained and one for normal build

Michadelic commented on webapp/index-self-contained.html:
```diff
@@ -0,0 +1,32 @@
+			data-sap-ui-libs="sap.m"
+			data-sap-ui-theme="sap_belize"
+			data-sap-ui-compatVersion="edge"
+			xx-data-sap-ui-preloadLibCss="!sap.ui.core,sap.m"
```
this somehow does not work - library css does not load when i check out this change and build

Michadelic commented on webapp/index-self-contained.html:
```diff
@@ -0,0 +1,32 @@
+		</script>
+
+		<script>
+			sap.ui.getCore().attachInit(function() {
```
better to not use inline JS anymore, i created an update on your pull

Michadelic committed:
made tests csp cpmpliant and async

Michadelic committed:
added -a to fix library CSS issues

Michadelic committed:
harmonized opa config

Michadelic committed:
moved shell from index to xml view

Michadelic committed:
bumped minVersion to 1.60 for async and component support compatibility

Michadelic reviewed (approved):
Very cool to showcase the self-contained build here in UI5Lab - thanks for this idea!

Michadelic commented on webapp/indexSelfContained.html:
```diff
@@ -0,0 +1,23 @@
+		data-sap-ui-libs="sap.m"
+		data-sap-ui-theme="sap_belize"
+		data-sap-ui-compatVersion="edge"
+		xx-data-sap-ui-preloadLibCss="!sap.ui.core,sap.m"
```
this line does not seem to work - at least for me, library CSS is still loaded for all 3 libs individually

matz3 replied:
That's because it starts with `xx-` which so to say invalidates the configuration. This seems to come from https://github.com/SAP/openui5-sample-app/blob/07f901c3146782d4cfad4b4ab3837df0405c8141/webapp/index-self-contained.html#L17 which also as a link tag commented out as bundling of CSS is not part of the UI5 Tooling, yet see https://github.com/SAP/ui5-builder/issues/91

Michadelic replied:
alright, let's ignore and keep it for now then

matz3 replied:
I would opt for removing them, as it's confusing and doesn't add value. Once this feature is available, the file anyway needs to be adopted.

Michadelic commented on webapp/Component.js:
```diff
@@ -1,7 +1,8 @@
 	"sap/ui/Device",
-	"ui5lab/app/GeometryApp/model/models"
+	"ui5lab/app/GeometryApp/model/models",
+	"sap/ui/core/ComponentSupport" // needed for self-contained build
```
known issue:
https://github.com/SAP/ui5-builder/issues/92
